### PR TITLE
fix(test): Add Labels tests

### PR DESCRIPTION
**Pull request by jarifibrahim**
Test results - http://sprunge.us/IQRU

### CODE REVIEW

**Comment by jarifibrahim** [comment]
@sanbornsen @nimishamukherjee The build failed due to some network issue. Can you please re-trigger the build?

File: runtime/src/tests/work-item/work-item-list/labels.spec.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  it('Verify Create New Label', function(){
+    detailPage = page.clickWorkItemTitle(page.firstWorkItem, firstitem.id);
+    detailPage.clickAddLabelButton();
+    let origLabelCount
+    detailPage.labelsCount.then(function(count){
+      origLabelCount = count
+    });
+    detailPage.clickCreateLabelButton();
+    detailPage.setLabelName(newLabelTitle);
+    detailPage.clickLabelCheck();
```
Comment (rgarg1) [inline runtime/src/tests/work-item/work-item-list/labels.spec.js]: checkbox

File: runtime/src/tests/work-item/work-item-list/labels.spec.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    // Verify label count has increased by 1
+    detailPage.labelsCount.then(function(count){
+      expect(count).toBe(origLabelCount + 1);
+    });
+    // Verify label exists in the list
+    expect(detailPage.listOfLabels().getText()).toContain(detailPage.labelTitleFromIndex(1));
```
Comment (rgarg1) [inline runtime/src/tests/work-item/work-item-list/labels.spec.js]: getLabelTitleByindex(index)

Comment (rgarg1) [reply runtime/src/tests/work-item/work-item-list/labels.spec.js]: better, GetLabelTitleBytext(text)

File: runtime/src/tests/work-item/work-item-list/labels.spec.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    page = new WorkItemListPage(true);
+    testSupport.setTestSpace(page);
+  });
+
+  it('Verify Add label button exists', function(){
+    detailPage = page.clickWorkItemTitle(page.firstWorkItem, firstitem.id);
+    expect(detailPage.addLabelButton.isPresent()).toBeTruthy();
+  });
+
+  it('Verify Add label button is clickable', function(){
```
Comment (rgarg1) [inline runtime/src/tests/work-item/work-item-list/labels.spec.js]: Combine this test with the previous one

File: runtime/src/tests/work-item/work-item-list/labels.spec.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    detailPage.clickAddLabelButton();
+    detailPage.selectLabel(1);
+    detailPage.selectLabel(3);
+
+    expect(detailPage.attachedLabels().getText()).toContain(detailPage.labelTitleFromIndex(3));
+    expect(detailPage.attachedLabels().getText()).toContain(detailPage.labelTitleFromIndex(1));
+    detailPage.clickLabelClose();
+  });
+
+  it('Verify removing existing label', function(){
```
Comment (rgarg1) [inline runtime/src/tests/work-item/work-item-list/labels.spec.js]: Specify by Unchecking the checkbox

File: runtime/src/tests/work-item/work-item-list/labels.spec.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    detailPage.clickAddLabelButton();
+    detailPage.clickCreateLabelButton();
+    detailPage.setLabelName(newLabelTitle);
+    detailPage.clickLabelCheck();
+    detailPage.selectLabel(1);
+    expect(detailPage.attachedLabels().getText()).toContain(newLabelTitle);
+    detailPage.clickLabelClose();
+  });
+
+  it('Verify removing new label', function(){
```
Comment (rgarg1) [inline runtime/src/tests/work-item/work-item-list/labels.spec.js]: Specify by clicking on the X

Comment (rgarg1) [reply runtime/src/tests/work-item/work-item-list/labels.spec.js]: Add a specific test to remove label by clicking the X button

File: runtime/src/tests/work-item/work-item-list/labels.spec.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    detailPage = page.clickWorkItemTitle(page.firstWorkItem,firstitem.id);
+    detailPage.clickAddLabelButton();
+    detailPage.selectLabel(defaultAttchedLabelIndex);
+    detailPage.clickLabelClose();
+
+    // Verify Label is removed (in detail page)
+    expect(detailPage.labelTitleFromIndex(defaultAttchedLabelIndex)).toBe("");
+  });
+
+  it('Verify adding new labels', function(){
```
Comment (rgarg1) [inline runtime/src/tests/work-item/work-item-list/labels.spec.js]: Eliminate the next test and check for label availability in the list view in this test itself. This makes the tests faster and eliminated redundancy.

**Comment by jarifibrahim** [comment]
@rgarg1 @ldimaggi Updated PR

**Comment by fabric8cd** [comment]
@jarifibrahim fabric8/fabric8-ui:SNAPSHOT-PR-2172-2 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2172-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@jarifibrahim fabric8/fabric8-ui:SNAPSHOT-PR-2172-3 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2172-fabric8-planner.badger.fabric8.io

**Comment by rgarg1** [comment]
@jarifibrahim Approved!

**Comment by fabric8cd** [comment]
@jarifibrahim fabric8/fabric8-ui:SNAPSHOT-PR-2172-5 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2172-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@jarifibrahim fabric8/fabric8-ui:SNAPSHOT-PR-2172-6 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2172-fabric8-planner.badger.fabric8.io

**Pull request merged**